lib/facility_factory.rb

```diff
@@ -0,0 +1,54 @@
+
+
+class FacilityFactory
+
+  def initialize
+  end
+
+  def create_facility(input_api)
+    co_total_facility_array = []
+    ny_total_facility_array = []
+    mo_total_facility_array = []
```

Comment:
You don't necessarily need separate arrays for each data set that could come in. This method will get run for only one set of data at a time, so just having one placeholder for `facilities` should be plenty. You'd just shovel all new facilities made into `facilities` rather than a specific array.